src/components/passphraseCreation/index.test.js

```diff
@@ -0,0 +1,113 @@
+  });
+
+  afterEach(() => {
+    passphraseUtils.generatePassphrase.restore();
+  });
+
+  describe('Using laptop', () => {
+    beforeEach(() => {
+      stub(checkDevice, 'default').returns(false);
+
+      wrapper = mount(<PassphraseCreation {...props} >
+        <CreateFirst t={() => {}}/>
+      </PassphraseCreation>);
+    });
+
+    afterEach(() => {
+      checkDevice.default.restore();
+    });
+
+    it('sets the next step once the movement is completed', () => {
+      expect(wrapper.find(CreateFirst)).to.have.prop('percentage', 0);
+      expect(wrapper.find(CreateFirst)).to.have.prop('hintTitle', 'by moving your mouse.');
+      expect(wrapper.find(CreateFirst)).to.have.prop('address', null);
+      expect(wrapper.find(CreateFirst)).to.have.prop('step', 'generate');
+      expect(wrapper.find(CreateFirst)).to.not.have.prop('passphrase');
+
+      for (let i = 0; i < 250; i++) {
+        events.mousemove({ pageX: 200 * (i % 2), pageY: 200 * (i % 2) });
+      }
+
+      wrapper.update();
+      expect(wrapper.find(CreateFirst)).to.have.prop('percentage').greaterThan(100);
+      expect(wrapper.find(CreateFirst)).to.have.prop('hintTitle', 'by moving your mouse.');
+      expect(wrapper.find(CreateFirst)).to.have.prop('address');
+      expect(wrapper.find(CreateFirst)).to.have.prop('step', 'info');
+      expect(wrapper.find(CreateFirst)).to.have.prop('passphrase', 'wagon stock borrow episode laundry');
+    });
+  });
+
+  describe('Using mobile device', () => {
+    beforeEach(() => {
+      window.addEventListener = (name, event) => {
+        events[name] = event;
+      };
+      stub(checkDevice, 'default').returns(true);
+    });
+
+    afterEach(() => {
+      checkDevice.default.restore();
+    });
+
+    it('returns child element with correct properties', () => {
+      wrapper = mount(<PassphraseCreation {...props} >
+        <CreateFirst t={() => {}}/>
+      </PassphraseCreation>);
+
+      expect(wrapper.find(CreateFirst)).to.have.prop('percentage', 0);
+      expect(wrapper.find(CreateFirst)).to.have.prop('hintTitle', 'by tilting your device.');
+      expect(wrapper.find(CreateFirst)).to.have.prop('address', null);
+      expect(wrapper.find(CreateFirst)).to.have.prop('step', 'generate');
+      expect(wrapper.find(CreateFirst)).to.not.have.prop('passphrase');
+    });
+
+    it('gets triggered and generates the seed on device tilt', () => {
+      wrapper = mount(<PassphraseCreation {...props} ><div></div></PassphraseCreation>);
+
+      const date = new Date();
+      const time = new Date(date.setSeconds(date.getSeconds() - 5));
+      // fake time difference
+      wrapper.instance().lastCaptured.time = time;
+      wrapper.instance().addEventListener();
+
+      // now tilt device without breaks
+      for (let i = 0; i < 250; i++) {
+        events.devicemotion({ rotationRate: { alpha: 200 * i, beta: 200 * i } });
+      }
+
+      expect(passphraseUtils.generatePassphrase).to.not.have.been.calledWith();
```

Comment:
I thought we concluded that it will be asserting how many times `generateSeed` was called (sinon callCount).